File: php/bubble_sort.php
<?php
/**
*bubble sort php
**/

	function bubble_sort() 
	{
		$input_data = [7,5,9,3,6,2,11,8];
		$t;

		for ($i=0; $i<count($input_data)-1; $i++) {
			for ($j=0; $j<count($input_data)-1-$i; $j++) {
				echo $i,$j;
				//bubble
				if ($input_data[$j] > $input_data[$j+1]) {
					$t = $input_data[$j];
					$input_data[$j] = $input_data[$j+1];
					$input_data[$j+1] = $t;
				}

				echo json_encode($input_data)."\n";
			} 
		}	

		//print_r($input_data);

	}


	bubble_sort();

File: python/bubble_sort.py
#!/usr/bin/env python
# coding: utf-8
# bubble sort python

def bubble_sort():
	#input_data = [7,5,9,3,6,2,11,8]
	input_data = [3,2,1]
	t = 0
	
	for index,item in enumerate(input_data,start=0):
		for i in xrange(len(input_data)-index-1):
			if(input_data[i] > input_data[i+1]):
				print index,i
				t = input_data[i]
				input_data[i] = input_data[i+1]
				input_data[i+1] = t

				print input_data	




bubble_sort()
